metadata/metadata.go

```diff
@@ -14,19 +14,28 @@
 
 package metadata
 
-import "errors"
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/mendersoftware/log"
+)
+
+type Validater interface {
+	Validate() error
+}
 
 var ErrInvalidInfo = errors.New("invalid artifacts info")
 
 type MetadataInfo struct {
 	Format  string `json:"format"`
-	Version string `json:"version"`
+	Version int    `json:"version"`
 }
 
-type MetadataInfoJSON string
-
 func (m MetadataInfo) Validate() error {
-	if len(m.Format) == 0 || len(m.Version) == 0 {
+	if len(m.Format) == 0 || m.Version == 0 {
```

Comment:
Not here. This is supposed to be "general" check to validate content. We will check for exact values while we will read this in client.